go/tools/builders/stdliblist.go

```diff
@@ -145,6 +146,42 @@ func flatPackageForStd(execRoot string, pkg *goListPackage) *flatPackage {
 	return newPkg
 }
 
+// In Go 1.18, the standard library started using go:embed directives.
+// When Bazel runs this action, it does so inside a sandbox where GOROOT points
+// to an external/go_sdk directory that contains a symlink farm of all files in
+// the Go SDK.
+// If we run "go list" with that GOROOT, this action will fail because those
+// go:embed directives will refuse to include the symlinks in the sandbox.
+//
+// To work around this, cloneRoot creates a copy of external/go_sdk into a new
+// directory "root" while retaining its path relative to the root directory.
+// So "$OUTPUT_BASE/external/go_sdk" becomes
+// "$OUTPUT_BASE/root/external/go_sdk".
+// This ensures that file paths in the generated JSON are still valid.
+//
+// cloneRoot returns the new root directory and the new GOROOT we should run
+// under.
+func cloneRoot(execRoot, goroot string) (newRoot string, newGoroot string, err error) {
+	relativeGoroot, err := filepath.Rel(abs(execRoot), abs(goroot))
+	if err != nil {
+		// GOROOT has to be a subdirectory of execRoot.
+		// Otherwise we're breaking the sandbox.
+		return "", "", fmt.Errorf("GOROOT %q is not relative to execution root %q: %v", goroot, execRoot)
+	}
+
+	newRoot = filepath.Join(execRoot, "root")
```

Comment:
I am a bit nervous about writing things to execRoot. It's supposed to contain a symlink tree intended to be read-only. This may have unexpected result if there happen to be a `root` directory under `execRoot`. Can you copy it to `ioutil.TempDir`?